lib/portlayer/storage/image.go

```diff
@@ -72,9 +72,9 @@ type ImageStorer interface {
 	// parent - The parent image to create the new image from.
 	// ID - textual ID for the image to be written
 	// meta - metadata associated with the image
+	// sum - expected sha266 sum of the image content.
 	// r - the image tar to be written
-	WriteImage(ctx context.Context, parent *Image, ID string, meta map[string][]byte, r io.Reader) (*Image,
-		error)
+	WriteImage(ctx context.Context, parent *Image, ID string, meta map[string][]byte, sum string, r io.Reader) (*Image, error)
```

Comment:
just a general comment, it doesn't need to be addressed at all. 

I personally like using structs as params approach. We could create a WriteImageOptions struct with those field and start passing that. This way we don't end up changing function signatures at all.